src/feature/gui/gui-runner.c

```diff
@@ -656,7 +657,10 @@ void mGUIRun(struct mGUIRunner* runner, const char* path) {
 				runner->core->reset(runner->core);
 				break;
 			case RUNNER_SAVE_STATE:
-				mCoreSaveState(runner->core, item->data.v.u >> 16, SAVESTATE_SCREENSHOT | SAVESTATE_SAVEDATA | SAVESTATE_RTC | SAVESTATE_METADATA);
+				struct mGUIBackground* gbaBackground = (struct mGUIBackground*) stateSaveMenu.background;
+				unsigned stateId = item->data.v.u >> 16;
+				gbaBackground->screenshotId = stateId | SCREENSHOT_STALE;
```

Comment:
I originally had an `if` check here that would check if the `screenshotId` was referring to the `stateId` in question. However, I believe it is impossible for it not to because we would have [updated `screenshotId`](https://github.com/mgba-emu/mgba/blob/master/src/feature/gui/gui-runner.c#L164-L169) in the draw loop when we [changed state IDs](https://github.com/mgba-emu/mgba/blob/master/src/feature/gui/gui-runner.c#L130)... does this sound correct?

Reply:
That does, yeah, in which case you can just |= the value with the flag instead of wholesale assigning it--which avoids the need for the `stateId` variable entirely.